README.md

```diff
@@ -28,93 +28,54 @@ This SDK uses [Semantic Versioning 2.0.0](https://semver.org).
 
-Usage is simple and revolves around one type, conveniently named `UserLeap`. Create one instance of this and keep it around while your application runs. The most obvious place for this is in your `ApplicationDelegate`, but do what's appropriate for your application.
+Usage is simple and revolves around a singleton, conveniently named `UserLeap`. The singleton must be configured before it can be used. The most obvious place for this is in your `ApplicationDelegate`, but do what's appropriate for your application.
 
 ```swift
-let leap = UserLeap(environment: "YOUR_ENVIRONMENT_ID_HERE")
+UserLeap.shared.configure(environment: "YOUR_ENVIRONMENT_ID_HERE")
 ```
 
 ### Setting properties
 
 Setting the user identifier helps to provide a consistent experience across platforms.
 
 ```
-leap.setUserIdentifier("myUserId")
+UserLeap.shared.setUserIdentifier("myUserId")
 ```
 
-Setting the email address allows us to deliver certain surveys over email.
+If you set the user identifier, it's important you also clear it when the user logs out. This will prevent new activity being associated with the wrong user.
 
 ```
-leap.setEmailAddress("example@email.com")
-```
-
-## Presenting a Survey
-
-When a survey is available, it's up to you to present it. The SDK provides a few mechanisms for determining if a survey is available for the current user.
-
-```swift
-  leap.presentSurvey(fromViewController: myViewController)
-```
-
-### Use Ready Callback
-
-You can provide a callback to the initializer that will be called whenever the `readyState` changes.
-
-```swift
-let leap = UserLeap(environment: "YOUR_ENVIRONMENT_ID_HERE") { leap in
-  if leap.readyState == .surveyReady {
-    leap.presentSurvey(fromViewController: myViewController)
-  }
-}
+UserLeap.shared.logout()
 ```
 
-### Use the NotificationCenter
-
-You can register for notifications about changes to the `readyState`.
+Setting the email address allows us to deliver certain surveys over email.
 
-```swift
-NotificationCenter.default.addObserver(forName: UserLeap.ReadyStateDidChangeNotification, object: nil, queue: .main) { [weak self] notification in
-  guard let self = self, let leap = notification.object as? UserLeap else { return }
-  if leap.readyState == .surveyReady {
-    leap.presentSurvey(fromViewController: self)
-  }
-}
 ```
-
-Alternatively, you can listen for notifications only when the survey becomes ready to present:
-
-```swift
-NotificationCenter.default.addObserver(forName: UserLeap.SurveyReadyNotification, object: nil, queue: .main) { [weak self] notification in
-  guard let self = self, let leap = notification.object as? UserLeap else { return }
-  leap.presentSurvey(fromViewController: self)
-}
+UserLeap.shared.setEmailAddress("example@email.com")
 ```
 
+There are various other properties you may want to set. These help us deliver the most appropriate surveys based on the user's engagement. For more details, [see the online documentation](https://docs.userleap.com/installation).
```

Comment:
This is good but can we also include details about the function name and parameters as well